Allow idempotent scheduler plugins to be registered via the RPC

jacobtomlinson opened the pull request:
The `Scheduler.add_plugin` has an `idempotent` flag which can be set to `True` to ensure a plugin is only registered once the first time. The default behaviour is to overwrite the plugin with new ones from subsequent attempts.

This flag is not exposed to the `register_scheduler_plugin` RPC method. This PR simply passes that flag through.

jrbourbeau reviewed (commented):
Thanks @jacobtomlinson. Exposing the `idempotent` keyword seems totally reasonable. Would you mind adding a small test for this change?

Also, this exposes an issue where we'll still call the plugin's `.start()` method even if `idempotent=True`. Perhaps we should have some extra logic to `return` early in that case?

jacobtomlinson commented:
Thanks for the review @jrbourbeau.

To address the starting of plugins I added another check an even earlier return. Also added tests for idempotent and non-idempotent plugins.
